src/hooks.ts

```diff
@@ -0,0 +1,79 @@
+import { useState } from 'react';
+import useDataStore from './stores/data';
+import { Columns, StandardizedVarible } from './utils/types';
+
+export const useColumnUpdates = () => {
+  const updateColumnDescription = useDataStore((state) => state.updateColumnDescription);
+  const updateColumnDataType = useDataStore((state) => state.updateColumnDataType);
+  const updateColumnStandardizedVariable = useDataStore(
+    (state) => state.updateColumnStandardizedVariable
+  );
+
+  const handleDescriptionChange = (columnId: string, newDescription: string | null) => {
+    updateColumnDescription(columnId, newDescription);
+  };
+
+  const handleDataTypeChange = (
+    columnId: string,
+    newDataType: 'Categorical' | 'Continuous' | null
+  ) => {
+    updateColumnDataType(columnId, newDataType);
+  };
+
+  const handleStandardizedVariableChange = (
+    columnId: string,
+    newStandardizedVariable: StandardizedVarible | null
+  ) => {
+    updateColumnStandardizedVariable(columnId, newStandardizedVariable);
+  };
+
+  return {
+    handleDescriptionChange,
+    handleDataTypeChange,
+    handleStandardizedVariableChange,
+  };
+};
+
+export const useTablePagination = (initialRowsPerPage: number) => {
+  const [page, setPage] = useState(0);
```

Comment:
like mentioned above, we can probably have the hook deal with the initialization of `page`, and then keep the setter local